libs/ledgerjs/packages/hw-app-exchange/src/SwapUtils.test.ts

```diff
@@ -0,0 +1,56 @@
+import { decodePayloadProtobuf } from "./SwapUtils"; // Asegúrate de proporcionar la ruta correcta
+
+describe("decodePayloadProtobuf function", () => {
+  test("should decode NewTransactionResponse correctly with device transaction id", async () => {
+    const binaryPayload =
+      "0a2a3078393736633339353463356462626633396135393135313064623332643266386363323235323830371a2a3078636361454263423338373661373561623945393639373530353861413735343633373733303239632a2a626331717164796b647738753336796864736c657477737976347865393573333735716a6a7934676b303a0345544842034254434a10000000000000000001118f178fb48000521000000000000000000000000000080e355a0a5552554f4b5149424f42";
```

Comment:
I think that the params `hexBinaryPayload` receive from the ExchangeSDK is a base64 encoded payload.
However, what is coming from native swap, I am not sure. It might be a simple "hex binary".

Reply:
it has been tested e2e with Changelly